evals/evaluation/crag_eval/README.md

```diff
@@ -0,0 +1,123 @@
+**Please note**: This is an example. You can build your own agent systems using OPEA components, then expose your own systems as an endpoint for this benchmark.</br>
+To launch the agent in our AgentQnA example, open another terminal and build images and launch agent system there.
+1. Build images
+```
+export $WORKDIR=<your-work-directory>
+cd $WORKDIR
+git clone https://github.com/opea-project/GenAIExamples.git
+cd GenAIExamples/AgentQnA/tests/
+bash 1_build_images.sh
+```
+2. Start retrieval tool
+```
+bash 2_start_retrieval_tool.sh
+```
+3. Ingest data into vector database and validate retrieval tool
+```
+# As an example, we will use the index_data.py script in AgentQnA example.
+# You can write your own script to ingest data.
+# As an example, We will ingest the docs of the music domain.
+# We will use the crag-eval docker container to run the index_data.py script.
+# The index_data.py is a client script.
+# it will send data-indexing requests to the dataprep server that is part of the retrieval tool.
+# So you need to switch back to the terminal where the crag-eval container is running.
+cd $WORKDIR/GenAIExamples/AgentQnA/retrieval_tool/
+python3 index_data.py --host_ip $host_ip --filedir ${WORKDIR}/datasets/crag_docs/ --filename crag_docs_music.jsonl
+```
+4. Launch and validate agent endpoint
+```
+# Go to the terminal where you launched the AgentQnA example
+cd $WORKDIR/GenAIExamples/AgentQnA/tests/
+bash 4_launch_and_validate_agent.sh
+```
+
+## Run CRAG benchmark
+Once you have your agent system up and running, the next step is to generate answers with agent. Change the variables in the script below and run the script. By default, it will run a sampled set of queries in music domain.
+```
+# Come back to the interactive crag-eval docker container
+cd $WORKDIR/GenAIEval/evals/evaluation/crag_eval/run_benchmark
+bash run_generate_answer.sh
+```
+
+## Use LLM-as-judge to grade the answers
+1. Launch llm endpoint with HF TGI: in another terminal, run the command below. By default, `meta-llama/Meta-Llama-3-70B-Instruct` is used as the LLM judge.
+```
+cd llm_judge
+bash launch_llm_judge_endpoint.sh
+```
+2. Validate that the llm endpoint is working properly.
+```
+export host_ip=$(hostname -I | awk '{print $1}')
+curl ${host_ip}:8085/generate_stream \
+    -X POST \
+    -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":20}}' \
+    -H 'Content-Type: application/json'
+```
+And then go back to the interactive crag-eval docker, run command below.
+```
+# Inside the crag-eval container
+cd $WORKDIR/GenAIEval/evals/evaluation/crag_eval/run_benchmark/llm_judge/
+python3 test_llm_endpoint.py
+```
+3. Grade the answer correctness using LLM judge. We use `answer_correctness` metrics from [ragas](https://github.com/explodinggradients/ragas/blob/main/src/ragas/metrics/_answer_correctness.py).
```

Comment:
hi, minmin, what about other metrics that we discussed?

Reply:
@lkk12014402 Unfortunately, the agent endpoint does not return the retrieved context, so cannot use metrics that require context.

Reply:
ok